### PR TITLE
Upgrade to Go 1.18

### PR DESCRIPTION
**Pull request by alexhung**
- Create own `Set` type for `Difference()` and `Intersection()` funcs
- Cleanup/replace deprecated code due to new `Set` and generics
- Update `getTestResty()` to use either `PROJECT_*` or `JFROG_*` env vars

### CODE REVIEW

**Review by danielmkn — APPROVED** [review]
All tests passed

**Review by maheshjfrog — APPROVED** [review]

File: pkg/project/resource_project_role.go [hunk 1]
```diff
@@ -190,33 +170,36 @@ var updateRoles = func(projectKey string, terraformRoles []Role, m interface{})
...
+
+	rolesToBeAdded := terraformRolesSet.Difference(projectRolesSet)
 	log.Printf("[TRACE] rolesToBeAdded: %+v\n", rolesToBeAdded)
 
-	rolesToBeUpdated := intersection(rolesToEquatables(terraformRoles), rolesToEquatables(projectRoles))
+	rolesToBeUpdated := terraformRolesSet.Intersection(projectRolesSet)
 	log.Printf("[TRACE] rolesToBeUpdated: %+v\n", rolesToBeUpdated)
 
-	rolesToBeDeleted := difference(rolesToEquatables(projectRoles), rolesToEquatables(terraformRoles))
+	rolesToBeDeleted := projectRolesSet.Difference(terraformRolesSet)
```
Comment (chb0github) [inline pkg/project/resource_project_role.go:182]: Not sure how you implemented this, but pretty clever. Or, is it part of 1.18

Comment (alexhung) [reply pkg/project/resource_project_role.go:182]: It's partly 1.18. But not as much as I'd like it can be.

Comment (alexhung) [reply pkg/project/resource_project_role.go:182]: I created our own `Set` https://github.com/jfrog/terraform-provider-project/pull/36/files#diff-f0b31058c422450c709fefe032c3745ab2ae2d4f1a8719cfffdaeb6ead6bfc9f

**Review by maheshjfrog — APPROVED** [review]

**Review by chb0github — COMMENTED** [review]
Approve. But, I can't now

**Pull request merged**